__tests__/ArtifactUploader.test.ts

```diff
@@ -134,4 +196,15 @@ describe('ArtifactUploader', () => {
     function mockListWithoutAssets() {
         listArtifactsMock.mockResolvedValue({ data: [] })
     }
+
+    function mockUploadArtifact(status: number = 200, failures: number = 0) {
+        uploadMock.mockImplementation(() => new Promise((resolve, reject) => {
+            if (failures > 0) {
+                failures--
```

Comment:
I think we can remove this line, it doesn't look like `failures` is used again within this scope.

Reply:
It is just used by L202. The variable is caputured by reference.

I'll rewrite this part to make it more readable.